cedar-lean/Cedar/Spec/ExtFun.lean

```diff
@@ -37,6 +37,8 @@ inductive ExtFun where
   | isLoopback
   | isMulticast
   | isInRange
+  | unknown             ----- Partial Evaluation Functions ----
+  | error
```

Comment:
What is the `error` extension for?

If possible, I think it's be better to keep the partial evaluation functionality contained to `Cedar/Partial` rather than let it leak into `Cedar/Spec`.

Reply:
Can we think of some other solution here?  This error function is not part of the language spec itself, and it would be a pity to make the spec messier for a specific feature like PE.

Reply:
It will be part of the spec once we release partial evaluation no? As in it will be a valid extension function?

Reply:
Thinking about it more, I think this function is quite problematic.  Specifically, what is its type? If we can't assign a type to it, then PE will produce ill-typed expressions even when given well-typed input. 

In other words, PE should preserve correct types, just like it preserves correct evaluation semantics (for non-erroring executions).

Reply:
If we don't like this solution then we have to be ok with not evaluating branches to errors, which is maybe fine.